### PR TITLE
Allow merge PRs with an exact hash

### PR DESCRIPTION
**Pull request by rwestberg**
Make it possible to create a merge-style PR with commits that are not already present in another existing repository. In this case, the hash of the commit to be merged must be specified, to allow for proper rebase / squash of the merge commit itself.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.java.net/census#ihse) (@magicus - Committer) ⚠️ Review applies to 6d9ea6295040578078e75731d56c5d2964127a9b
 * [Erik Helin](https://openjdk.java.net/census#ehelin) (@edvbld - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/skara pull/1026/head:pull/1026`
`$ git checkout pull/1026`

### CODE REVIEW

File: bots/pr/src/test/java/org/openjdk/skara/bots/pr/MergeTests.java [hunk 1]
```diff
@@ -127,6 +127,154 @@ void branchMerge(TestInfo testInfo) throws IOException {
...
+            try (var tempCommits = pushedRepo.commits(masterHash.hex() + ".." + headHash.hex())) {
+                commits = tempCommits.stream()
+                                     .map(Commit::hash)
+                                     .collect(Collectors.toSet());
+            }
+            assertTrue(commits.contains(otherHash1));
+            assertTrue(commits.contains(otherHash2));
+            assertFalse(commits.contains(mergeHash));
+
+            // Author and committer should updated in the merge commit
```
Comment (magicus) [inline bots/pr/src/test/java/org/openjdk/skara/bots/pr/MergeTests.java:206]:
```suggestion
            // Author and committer should be updated in the merge commit
``

**Review by magicus — APPROVED** [review]
Looks good (minus the typo)

**Review by edvbld — COMMENTED** [review]
Looks good overall, really nice solution :100: Just two things I thought of:

- should we check that the the hash actually is an ancestor of the source branch? That check combined with the existing check that the hash is not an ancestor of the target branch would ensure that the hash only is in the source branch for the pull request. Would safe-guard against some extremely unlikely scenarios :smile: 
- can you make it possible via a bot configuration to restrict the users that can do merge style pull requests with a hash for a repository? I don't think all repositories are gonna want to allow everyone to use a `Merge <HASH>` kind of pull request.

Thanks!

**Review by edvbld — APPROVED** [review]
Looks good!

**Pull request closed without merging**